jerry-libc/jerry-libc-printf.c

```diff
@@ -210,23 +211,13 @@ libc_printf_write_d_i (FILE *stream, /**< stream pointer */
   switch (length)
   {
     case LIBC_PRINTF_ARG_LENGTH_TYPE_NONE:
+    case LIBC_PRINTF_ARG_LENGTH_TYPE_HH: /* char is promoted to int */
+    case LIBC_PRINTF_ARG_LENGTH_TYPE_H: /* short int is promoted to int */
     {
       value = (uintmax_t)va_arg (*args_list_p, int);
       break;
     }
 
-    case LIBC_PRINTF_ARG_LENGTH_TYPE_HH:
-    {
-      value = (uintmax_t)va_arg (*args_list_p, int); /* char is promoted to int */
-      break;
-    }
-
-    case LIBC_PRINTF_ARG_LENGTH_TYPE_H:
-    {
-      value = (uintmax_t)va_arg (*args_list_p, int); /* short int is promoted to int */
-      break;
-    }
-
```

Comment:
Me too. Still, as the table above shows, doing the merge manually does have an impact on the code size.

Generally, I think it is good practice not to write overly verbose, or suboptimal, or clone-full code and rely on the compiler to optimize it away. Since sometimes it fails. (Up to a certain point of course. Extreme brevity and incomprehensible optimizations can have a critical maintenance cost in the long run, but that's another story. And this is not the case here, IMHO.)